File: examples/ruby/ownership.rb
def take list
  p "#{list[0]}"
end

list = [10, 5, 4]
listb = list
listc = listb

# list.push(42)

p "#{list}"
p "#{listb}"
p "#{listc}"

take list
take list

# Running
# cd examples/ruby/
# ruby ownership.rb

File: examples/rust/ownership.rs
fn take(list: Vec<i32>) { println!("{:?}", list)}

fn main() {
    let a = vec![10, 5, 4];
    take(a); // Transfere o owner
    println!("{:?}", a); // ERRROUUU error: use of moved value: `a`
}

// Running
// cd examples/rust/
// rustc ownership.rs && ./ownership
